frontend/src/App.tsx

```diff
@@ -5,18 +5,28 @@ import {BrowserRouter, Route, Routes} from "react-router-dom";
 import SpotDetail from "./components/SpotDetail";
 import useSpots from "./hooks/useSpots";
 import SpotList from "./components/spot_list/SpotList";
+import BottomNavigationBar from "./components/BottomNavigationBar";
+import {Spot} from "./models/Spot";
 
 function App() {
 
     const {mapboxToken} = useAccessToken()
-    const {spots} = useSpots()
+    const {spots, addSpot} = useSpots()
+
+    function handleAddSpot(newSpot:Spot):Promise<void> {
+        return addSpot(newSpot).then(()=>{
```

Comment:
What's the purpose of `.then(()=>{        })`?
`addSpot(newSpot)` seems to return a Promise anyhow